src/server/system_services/bucket_server.js

```diff
@@ -1998,13 +2017,41 @@ function validate_replication(req) {
 }
 
 function normalize_replication(req) {
-    const replication_rules = req.rpc_params.replication_policy;
+    const replication_rules = req.rpc_params.replication_policy.rules;
 
-    const validated_replication = replication_rules.map(rule => {
+    const validated_replication_rules = replication_rules.map(rule => {
         const { destination_bucket } = rule;
         const dst_bucket = req.system.buckets_by_name && req.system.buckets_by_name[destination_bucket.unwrap()];
         return { ...rule, destination_bucket: dst_bucket._id };
     });
+
+    const validated_replication = { rules: validated_replication_rules };
+
+    if (req.rpc_params.replication_policy.log_replication_info) {
+        let validated_log_based_replication;
+        const logs_bucket = req.system.buckets_by_name &&
+            req.system.buckets_by_name[req.rpc_params.replication_policy.log_replication_info.logs_location.logs_bucket.unwrap()];
+        if (!logs_bucket) {
+            const source_bucket = find_bucket(req);
+            validated_log_based_replication = {
+                log_replication_info: { logs_location: { logs_bucket: source_bucket._id } }
```

Comment:
why doing it?

Reply:
its one of the default behaviour we agreed on, if logs_bucket is not provided assume the logs are in the source bucket

Reply:
so add it as a comment please here

Reply:
will do